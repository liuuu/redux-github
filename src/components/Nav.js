// @flow
import React from "react";
import { Link } from "react-router-dom";

import { Menu, Layout } from "antd";
import { Icon } from "antd";
import { Input } from "antd";

const { Header } = Layout;
class Nav extends React.Component {
  render() {
    return (
      <Layout>
        <Header>
          <Menu
            mode="horizontal"
            style={{ lineHeight: "64px", fontSize: "20px" }}
            theme="dark"
          >
            <Menu.Item><Link to="/"><Icon type="windows-o" /></Link></Menu.Item>
            <Menu.Item><Link to="/all">browse all</Link></Menu.Item>
            <Menu.Item style={{ float: "right" }}>About</Menu.Item>
          </Menu>
        </Header>
      </Layout>
    );
  }
}

export default Nav;
