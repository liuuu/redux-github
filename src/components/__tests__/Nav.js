import React from "react";
import Nav from "../Nav";
import { shallow, mount, render } from "enzyme";
import toJson from "enzyme-to-json";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { Menu } from "antd";

it("should match snapshot Nav", () => {
  const tree = render(
    <MemoryRouter>
      <Nav />
    </MemoryRouter>
  );
  expect(toJson(tree)).toMatchSnapshot();
});

it("should have 1 Menu component", () => {
  const component = mount(
    <MemoryRouter>
      <Nav />
    </MemoryRouter>
  );

  expect(component.find(Menu)).toHaveLength(1);
});
