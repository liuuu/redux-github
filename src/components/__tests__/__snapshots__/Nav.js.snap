// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot Nav 1`] = `
<div
  class="ant-layout"
>
  <div
    class="ant-layout-header"
  >
    <ul
      aria-activedescendant=""
      class="ant-menu ant-menu-horizontal  ant-menu-dark ant-menu-root"
      role="menu"
      style="line-height:64px;font-size:20px;"
      tabindex="0"
    >
      <li
        aria-selected="false"
        class="ant-menu-item"
        role="menuitem"
      >
        <a
          href="/"
        >
          <i
            class="anticon anticon-windows-o"
          />
        </a>
      </li>
      <li
        aria-selected="false"
        class="ant-menu-item"
        role="menuitem"
      >
        <a
          href="/all"
        >
          browse all
        </a>
      </li>
      <li
        aria-selected="false"
        class="ant-menu-item"
        role="menuitem"
        style="float:right;"
      >
        About
      </li>
    </ul>
  </div>
</div>
`;
